lib/puppet/configurer.rb

```diff
@@ -53,7 +65,7 @@ def initialize(factory = Puppet::Configurer::DownloaderFactory.new)
   end
 
   # Get the remote catalog, yo.  Returns nil if no catalog can be found.
-  def retrieve_catalog(query_options)
+  def retrieve_catalog(query_options, options)
```

Comment:
`retrieve_catalog` is potentially public, we should default this to an empty hash.